gemrb/GUIScripts/iwd/CharGen.py

```diff
@@ -292,6 +292,9 @@ def BackPress():
 		SkillsButton.SetState (IE_GUI_BUTTON_ENABLED)
 		SkillsButton.SetFlags (IE_GUI_BUTTON_DEFAULT, OP_OR)
 		SkillsState = 0
+		#remove all known spells and nullify the memorizable counts
+		Spellbook.RemoveKnownSpells (MyChar, IE_SPELL_TYPE_WIZARD, 1,9, 1)
+		Spellbook.RemoveKnownSpells (MyChar, IE_SPELL_TYPE_PRIEST, 1,7, 1)
```

Comment:
This only works if one returns to a step sooner than skills, so it needs to be moved. I suggest where the proficiency window is set up, since all classes will hit that.

Reply:
Thanks for suggestion, will do that, and test if that does the trick for all classes.

Reply:
That doesn't work for mage class, but I will figure out right place sooner or later. ;)

Reply:
it should — everyone gets to pick weapon proficiencies.

Reply:
Make sure the core MageMemorized and MageSpellBook get wiped too.

Reply:
Indeed, after adding 
```
GemRB.SetVar ("MageMemorized", 0)
GemRB.SetVar ("MageSpellBook", 0)
```
it worked. As I understand, adding `GemRB.SetVar ("SpellMask", 0)` won't harm either. Will create function in Spellbook for wiping spells, I found that snippet to remove spells in Baldur's Game chargen file.

Reply:
I'm not sure another function is needed. RemoveKnownSpells does all the heavy lifting, it's just that iwd chargen doesn't use all the shared code yet (OpenSpellSelectionWindow or something like that). Separate little project, believe me.

Reply:
I trust your opinion, as you are clearly GemRB guru. ;)